src/components/application_manager/src/request_controller.cc

```diff
@@ -364,9 +369,27 @@ void RequestController::onTimer() {
   SDL_AUTO_TRACE();
   SDL_DEBUG(
       "ENTER Waiting fore response count: " << waiting_for_response_.Size());
-  RequestInfoPtr probably_expired =
-      waiting_for_response_.FrontWithNotNullTimeout();
-  while (probably_expired && probably_expired->isExpired()) {
+  while(!stop_flag_) {
```

Comment:
@LuxoftAKutsan This is not rewrite, actually the timer logic is not changed here,  here add ```AutoLock``` to make it works better in multi-thread program. And during the test, I usually found ```TIME_OUT`` error based on the machine performance.